### PR TITLE
chore:  Overlay calendar - enabled by default if supported calendar installed

### LINKED ISSUE #14969: [CAL-3675] Overlay calendar - enabled by default if supported calendar is installed
Overlay calendar should be enabled by default if you have a calendar we support for overlaying. 

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3675](https://linear.app/calcom/issue/CAL-3675/overlay-calendar-enabled-by-default-if-supported-calendar-is-installed)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/features/bookings/Booker/Booker.tsx [hunk 1]
```diff
@@ -140,7 +144,16 @@ const BookerComponent = ({
       scrolledToTimeslotsOnce.current = true;
     }
   };
-
+  useEffect(() => {
+    if (connectedCalendars.length > 0 && hasSession) {
```
Comment (supalarry) [inline packages/features/bookings/Booker/Booker.tsx:148]: 'connectedCalendars' are passed by the 'calendars' prop, which is passed to 'Booker' by the 'BookerWebWrapper', which gets the 'calendars' using 'useCalendars' hook.

However, if we check 'useCalendars' hook, then fetching connected calendars is enabled only if the setting is set in state or 'overlayCalendar' exists as search param:
```
const { data, isPending } = trpc.viewer.connectedCalendars.useQuery(undefined, {
    enabled: !!calendarSettingsOverlay || Boolean(searchParams?.get("overlayCalendar")),
  });
```

But the newly added `useEffect` only sets 'overlayCalendar' to true, if there are connected calendars, so we have a closed circle - I tried testing this PR and even though I have connected a calendar visiting someone's booking page did nothing.

**Comment by hariombalhara** [comment]
@anikdhabal Could you add a loom? I don't understand what we are fixing here exactly.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Need loom

**Comment by supalarry** [comment]
> @anikdhabal Could you add a loom? I don't understand what we are fixing here exactly.

If you have logged into your app.cal.com and then try to book me i.cal.com/keith and click on some event type, then the overlay calendar toggle should be toggled by default if you have google calendar connected:
Screenshot 2024-07-08 at 09 43 19

File: packages/features/bookings/Booker/Booker.tsx (hunk 1 above)
Comment (anikdhabal) [reply packages/features/bookings/Booker/Booker.tsx]: Yeah, that's true. I will fix that

File: packages/platform/atoms/booker/BookerWebWrapper.tsx [hunk 2]
```diff
@@ -168,6 +168,13 @@ export const BookerWebWrapper = (props: BookerWebWrapperAtomProps) => {
     },
     [searchParams, pathname, router]
   );
+
+  useEffect(() => {
+    if (calendars.connectedCalendars.length > 0) {
```
Comment (supalarry) [inline packages/platform/atoms/booker/BookerWebWrapper.tsx:173]: Isn't the only supported overlay calendar google calendar now? Aka if you install MS outlook and the switch in this case is enabled, would it work?

**Pull request closed without merging**